src/test/java/org/openelisglobal/pathology/PathologySampleServiceTest.java

```diff
@@ -0,0 +1,116 @@
+package org.openelisglobal.pathology;
+
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.util.Collections;
+import java.util.List;
+import org.dbunit.DatabaseUnitException;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.openelisglobal.BaseWebContextSensitiveTest;
+import org.openelisglobal.program.controller.pathology.PathologySampleForm;
+import org.openelisglobal.program.service.PathologyBlockService;
+import org.openelisglobal.program.service.PathologySampleService;
+import org.openelisglobal.program.valueholder.pathology.PathologyBlock;
+import org.openelisglobal.program.valueholder.pathology.PathologySample;
+import org.openelisglobal.systemuser.service.SystemUserService;
+import org.openelisglobal.systemuser.valueholder.SystemUser;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class PathologySampleServiceTest extends BaseWebContextSensitiveTest {
+
+    @Autowired
+    PathologySampleService pathologySampleService;
+
+    @Autowired
+    PathologyBlockService pathologyBlockService;
+
+    @Autowired
+    SystemUserService systemUserService;
+
+    @Before
+    public void init() throws Exception {
+        executeDataSetWithStateManagement("testdata/pathology-sample.xml");
+    }
+
+    @Test
+    public void getAll_shouldReturnAllExistingPathologySamples() {
+        Assert.assertEquals(Integer.parseInt("2"), pathologySampleService.getAll().size());
+    }
+
+    @Test
+    public void getWithStatus_shouldReturnPathologySampleWhenGivenStatus() {
+        List<PathologySample> pathologySamples = pathologySampleService
+                .getWithStatus(List.of(PathologySample.PathologyStatus.GROSSING));
+
+        Assert.assertNotNull(pathologySamples);
+        Assert.assertEquals(PathologySample.PathologyStatus.GROSSING, pathologySamples.get(0).getStatus());
+        Assert.assertEquals("John", pathologySamples.get(0).getTechnician().getFirstName());
+    }
+
+    @Test
+    public void searchWithStatusAndTerm_shouldSearchGivenStatusAndSearchTerm() {
+        List<PathologySample> pathologySamples = pathologySampleService
+                .searchWithStatusAndTerm(List.of(PathologySample.PathologyStatus.GROSSING), "");
+
+        Assert.assertNotNull(pathologySamples);
+        Assert.assertEquals(PathologySample.PathologyStatus.GROSSING, pathologySamples.get(0).getStatus());
+        Assert.assertEquals("John", pathologySamples.get(0).getTechnician().getFirstName());
+    }
+
+    @Test
+    public void assignTechnician_shouldAssignTechnicianToPathologySample() throws SQLException, DatabaseUnitException {
+        SystemUser curUser = systemUserService.getUserById("1004");
+        pathologySampleService.assignTechnician(2, curUser, "");
+
+        PathologySample pathologySample = pathologySampleService.get(1);
+
+        Assert.assertEquals(Integer.valueOf(1), pathologySample.getId());
```

Comment:
can you also assert that this pathologySample has the asscoiated pathologyBlocks inserted in the dataset

Reply:
done!